### PR TITLE
chore: hook into QT logs with qInstallMessageHandler

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #8296

This makes the QML logs finally show on Windows

Let me know if you want to keep the Linux and Mac logs as it was before and only use this type of logging on Windows.

It looks like this on Windows:

image

### CODE REVIEW

File: lib/src/DOtherSide.cpp [hunk 1]
```diff
@@ -174,6 +177,32 @@ void dos_qguiapplication_initialize_opengl()
...
+// This catches the QT and QML logs and outputs them.
+// This is necessary on Windows, because otherwise we do not get any logs at all
+void myMessageOutput(QtMsgType type, const QMessageLogContext &context, const QString &msg)
+{
+    QByteArray localMsg = msg.toLocal8Bit();
+    const char *file = context.file ? context.file : "";
+    const char *function = context.function ? context.function : "";
+    switch (type) {
+    case QtDebugMsg:
+        fprintf(stderr, "Debug: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line, function);
```
Comment (osmaczko) [inline lib/src/DOtherSide.cpp:189]:
```suggestion
        fprintf(stdout, "Debug: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line, function);
```

File: lib/src/DOtherSide.cpp [hunk 2]
```diff
@@ -174,6 +177,32 @@ void dos_qguiapplication_initialize_opengl()
...
+        break;
+    case QtInfoMsg:
+        fprintf(stderr, "Info: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line, function);
```
Comment (osmaczko) [inline lib/src/DOtherSide.cpp:192]:
```suggestion
        fprintf(stdout, "Info: %s (%s:%u, %s)\n", localMsg.constData(), file, context.line, function);
```

**Review by osmaczko — COMMENTED** [review]
IMO output format should be consistent between platforms, meaning the same handler is ok :+1:

File: lib/src/DOtherSide.cpp (hunk 1 above)
Comment (jrainville) [reply lib/src/DOtherSide.cpp:189]: stdout didn't work. I'll keep stderr for now 🤷

**Review by richard-ramos — APPROVED** [review]

**Review by iurimatias — APPROVED** [review]

**Pull request merged**